projects/fw_103_hw/src/main.c

```diff
@@ -0,0 +1,69 @@
+#include <stdbool.h>
+#include <stdio.h>
+
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+
+// GPIO address that we will allow us to read button data
+const GpioAddress potentiometer_addr_A6 = { // could be passed as context
+        .port = GPIO_PORT_A,
+        .pin = 6,
+};
+
+
+// function that is called each time the button is clicked
+static void prv_button_interrupt_callback(const GpioAddress *potentiometer_addr_B2, void *context) {
+
+    uint16_t data = 0;
+    StatusCode button_data = adc_read_raw_pin( potentiometer_addr_A6, &data);  // see what this takes in in the header file
+
+    LOG_DEBUG("Status Code: %d\n", button_data);
+    LOG_DEBUG("button data: %d\n", data);
+};
+
+int main(void) {
+    interrupt_init();
+    gpio_init();
+    gpio_it_init();
+    soft_timer_init();
+    adc_init(ADC_MODE_SINGLE);
+
+    // GPIO Address where button is pressed
+    GpioAddress potentiometer_addr_B2 = {
```

Comment:
This wouldn't be a potentiometer address, this would be a button address, so it would be good to label it as such